Automattic/wp-calypso
# Framework: Remove sites-list from billing-history

This PR removes `sites-list` from Billing History (`/me/purchases/billing`), and ES6ifies several things that standed in the way.

~~Blocked by~~ Complemented by #13094.

To test:
* Checkout this branch.
* Go to http://calypso.localhost:3000/me/purchases/billing
* Verify everything there works as expected and there are no regressions:
 * Both tables display the records properly.
 * Links are working
 * Search is working
 * Transaction filtering is working.
* Verify all works correctly with a clean Redux state.
